core/src/solve.jl

```diff
@@ -256,21 +301,88 @@ function formulate!(du::AbstractVector, basin::Basin, u::AbstractVector, t::Real
         drainage = basin.drainage[i]
         infiltration = reduction_factor * basin.infiltration[i]
 
-        du[i] += precipitation - evaporation + drainage - infiltration
+        du.storage[i] += precipitation - evaporation + drainage - infiltration
     end
     return nothing
 end
 
+function get_error(pid_control::PidControl, p::Parameters)::Vector{Float64}
+    (; basin) = p
+    (; node_id, listen_node_id) = pid_control
+
+    error = zero(pid_control.derivative)
+
+    for (i, id) in enumerate(node_id)
+        listened_node_id = listen_node_id[i]
+        has_index, listened_node_idx = id_index(basin.node_id, listened_node_id)
+        @assert has_index "Listen node $listened_node_id is not a basin."
+        target_level = basin.target_level[listened_node_idx]
+        error[i] = target_level - basin.current_level[listened_node_idx]
+    end
+
+    return error
+end
+
+function continuous_control!(
+    du::AbstractVector,
+    pid_control::PidControl,
+    p::Parameters,
+    integral_::Vector{Float64},
```

Comment:
It is not directly clear to me how `integral_` and `integral` are different. Could you explain, perhaps with a comment in the code?